wait NOT LGTM fails on lxc
```
--- FAIL: TestRunCreateVolumesInSymlinkDir (3.29s)
	docker_utils.go:604: "run -v /test/test test-volume-symlink" failed with errors: lxc-start: conf.c: mount_entry: 2049 Not a directory - failed to mount '/var/lib/docker/vfs/dir/717d0a8409d806b4a9c37493c072a0024733b4d6b856422b2e8e3e3f2cce47dd' on '/usr/local/lib/lxc/rootfs///test/test'
		lxc-start: conf.c: lxc_setup: 4163 failed to setup the mount entries for 'c5d72d8a8ab6a66d15f0b4963ae65d48331ee4a0185caa8308d8f0e3c853e2bf'
		lxc-start: start.c: do_start: 688 failed to setup the container
		lxc-start: sync.c: __sync_wait: 51 invalid sequence number 1. expected 2
		lxc-start: start.c: __lxc_start: 1080 failed to spawn 'c5d72d8a8ab6a66d15f0b4963ae65d48331ee4a0185caa8308d8f0e3c853e2bf'
		lxc-start: lxc_start.c: main: 342 The container failed to start.
		lxc-start: lxc_start.c: main: 346 Additional information can be obtained by setting the --logfile and --logpriority options.
```

--- a/daemon/volumes.go
+++ b/daemon/volumes.go
@@ -189,13 +189,13 @@ func (container *Container) parseVolumeMountConfig() (map[string]*Mount, error) 
 		if _, exists := container.Volumes[path]; exists {
 			continue
 		}
-		realpath, err := symlink.FollowSymlinkInScope(filepath.Join(container.basefs, path), container.basefs)
+		realPath, err := container.getResourcePath(path)
 		if err != nil {
 			return nil, fmt.Errorf("failed to evaluate the absolute path of symlink")
 		}
-		if stat, err := os.Stat(realpath); err == nil {
+		if stat, err := os.Stat(realPath); err == nil {
 			if !stat.IsDir() {
-				return nil, fmt.Errorf("file exists at %s, can't create volume there", realpath)
+				return nil, fmt.Errorf("file exists at %s, can't create volume there", realPath)
 			}
 		}
 

--- a/integration-cli/docker_cli_run_test.go
+++ b/integration-cli/docker_cli_run_test.go
@@ -477,37 +477,34 @@ func TestRunWithVolumesFromExited(t *testing.T) {
 	logDone("run - regression test for #4979 - volumes-from on exited container")
 }
 
-// Test create volume in a dirctory which is a symbolic link
+// Volume path is a symlink which also exists on the host, and the host side is a file not a dir
+// But the volume call is just a normal volume, not a bind mount
 func TestRunCreateVolumesInSymlinkDir(t *testing.T) {
+	testRequires(t, SameHostDaemon)
+	testRequires(t, NativeExecDriver)
 	defer deleteAllContainers()
-	// This test has to create a file on host
-	hostFile := "/tmp/abcd"
-	cmd := exec.Command("touch", hostFile)
-	if out, _, err := runCommandWithOutput(cmd); err != nil {
-		t.Fatalf("failed to create file %s on host: %v, output: %q", hostFile, err, out)
-	}
-	defer func() {
-		cmd := exec.Command("rm", "-f", hostFile)
-		if out, _, err := runCommandWithOutput(cmd); err != nil {
-			t.Fatalf("failed to remove file %s on host: %v, output: %q", hostFile, err, out)
-		}
-	}()
-	// create symlink directory /home/test link to /tmp
-	cmd = exec.Command(dockerBinary, "run", "--name=test", "busybox", "ln", "-s", "/tmp", "/home/test")
-	if out, _, err := runCommandWithOutput(cmd); err != nil {
-		t.Fatalf("failed to run container: %v, output: %q", err, out)
-	}
-	cmd = exec.Command(dockerBinary, "commit", "test", "busybox:test")
-	out, _, err := runCommandWithOutput(cmd)
+	name := "test-volume-symlink"
+
+	dir, err := ioutil.TempDir("", name)
 	if err != nil {
-		t.Fatalf("failed to commit container: %v, output: %q", err, out)
+		t.Fatal(err)
 	}
-	cleanedImageID := stripTrailingCharacters(out)
-	defer deleteImages(cleanedImageID)
-	// directory /home/test is link to /tmp, /home/test/abcd==/tmp/abcd
-	cmd = exec.Command(dockerBinary, "run", "-v", "/home/test/abcd", "busybox", "touch", "/home/test/abcd/Hello")
-	if out, _, err = runCommandWithOutput(cmd); err != nil {
-		t.Fatalf("failed to create volume in symlink directory: %v, output %q", err, out)
+	defer os.RemoveAll(dir)
+
+	f, err := os.OpenFile(filepath.Join(dir, "test"), os.O_CREATE, 0700)
+	if err != nil {
+		t.Fatal(err)
+	}
+	f.Close()
+
+	dockerFile := fmt.Sprintf("FROM busybox\nRUN mkdir -p %s\nRUN ln -s %s /test", dir, dir)
+	if _, err := buildImage(name, dockerFile, false); err != nil {
+		t.Fatal(err)
+	}
+	defer deleteImages(name)
+
+	if out, _, err := dockerCmd(t, "run", "-v", "/test/test", name); err != nil {
+		t.Fatal(err, out)
 	}
 
 	logDone("run - create volume in symlink directory")

